pkgs/os-specific/linux/kernel/generic.nix

```
@@ -131,12 +131,13 @@ let
 
     buildPhase = ''
       export buildRoot="''${buildRoot:-build}"
+      export CC=${stdenv.cc.targetPrefix}cc
+      export HOSTCC=${buildPackages.stdenv.cc.targetPrefix}cc
+      export HOSTCXX=${buildPackages.stdenv.cc.targetPrefix}c++
```

Because of the cc- and bintools-wrapper setup hooks, I think we can do
```suggestion
      export HOSTCC=$CC_FOR_BUILD
      export HOSTCXX=$CXX_FOR_BUILD
```

---

>   OBJCOPY=llvm-objcopy OBJDUMP=llvm-objdump READELF=llvm-readelf \
>   HOSTCC=clang HOSTCXX=clang++ HOSTAR=llvm-ar HOSTLD=ld.lld

Per the above I would also do `HOSTAR=$AR_FOR_BUILD` and `HOSTLD=$LD_FOR_BUILD`.

Putting that altogether it's

```suggestion
      export HOSTCC=$CC_FOR_BUILD
      export HOSTCXX=$CXX_FOR_BUILD
      export HOSTAR=$AR_FOR_BUILD
      export HOSTLD=$LD_FOR_BUILD
```